### ISSUE TITLE
rusoto_core build failure on MacOS 12.6.1 with Apple M1 Max chip

### ISSUE THREAD

**Issue by chris-bucchere-lark** [delta-io/delta-rs#925]
# Environment

**Delta-rs version**: latest from main

**Binding**: n/a

**Environment**:
- **Cloud provider**: n/a
- **OS**: MacOS 12.6.1 (Monterey) Apple M1 Max silicon
- **Other**:

***
# Bug

**What happened**: `cargo build` failed

**What you expected to happen**: `cargo build` should build the `rusoto_core` binary and continue with the rest of the build

**How to reproduce it**: `git clone https://github.com/delta-io/delta-rs.git`; `cargo build`

**More details**:

```
% cargo build
   Compiling pyo3-build-config v0.17.2
   Compiling zstd-safe v5.0.2+zstd.1.5.2
   Compiling zstd-sys v2.0.1+zstd.1.5.2
   Compiling lz4-sys v1.9.4
   Compiling object_store v0.5.1
   Compiling rusoto_core v0.48.0
   Compiling blake3 v1.3.1
   Compiling glibc_version v0.1.2
   Compiling openssl-src v111.22.0+1.1.1q
   Compiling pkg-config v0.3.25
error[E0252]: the name `tls` is defined multiple times
  --> /Users/bucchere/.cargo/registry/src/github.com-1ecc6299db9ec823/rusoto_core-0.48.0/src/lib.rs:25:5
   |
23 | use hyper_rustls as tls;
   |     ------------------- previous import of the module `tls` here
24 | #[cfg(feature = "native-tls")]
25 | use hyper_tls as tls;
   |     ^^^^^^^^^^^^^^^^ `tls` reimported here
   |
   = note: `tls` must be defined only once in the type namespace of this module
help: you can use `as` to change the binding name of the import
   |
25 | use hyper_tls as other_tls;
   |     ~~~~~~~~~~~~~~~~~~~~~~

   Compiling lz4 v1.24.0
   Compiling deltalake v0.4.1 (/Users/bucchere/lark/delta-rs/rust)
   Compiling atty v0.2.14
   Compiling glob v0.3.0
   Compiling termcolor v1.1.3
   Compiling zstd v0.11.2+zstd.1.5.2
error[E0599]: no function or associated item named `new` found for struct `hyper_rustls::HttpsConnector` in the current scope
   --> /Users/bucchere/.cargo/registry/src/github.com-1ecc6299db9ec823/rusoto_core-0.48.0/src/request.rs:224:41
    |
224 |         let connector = HttpsConnector::new();
    |                                         ^^^ function or associated item not found in `hyper_rustls::HttpsConnector<_>`

error[E0599]: no function or associated item named `new` found for struct `hyper_rustls::HttpsConnector` in the current scope
   --> /Users/bucchere/.cargo/registry/src/github.com-1ecc6299db9ec823/rusoto_core-0.48.0/src/request.rs:244:41
    |
244 |         let connector = HttpsConnector::new();
    |                                         ^^^ function or associated item not found in `hyper_rustls::HttpsConnector<_>`

   Compiling pin-project-internal v1.0.12
Some errors have detailed explanations: E0252, E0599.
For more information about an error, try `rustc --explain E0252`.
error: could not compile `rusoto_core` due to 3 previous errors
warning: build failed, waiting for other jobs to finish...
```

**Comment by wjones127** [comment]
Duplicate of #677.

We should fix this, I've just never been successful at it and tried a couple times.

The error originates from the aws/delta-checkpoint folder, so if you aren't working on that component you can simply build from one of the subdirectories instead and it should work.

**Comment by chris-bucchere-lark** [comment]
Thanks and sorry for the dupe.

**Comment by rtyler** [comment]
Related ticket is fixed, closing
